File: JavaScript/Gehoelz.js
/*
 * autor:   Olaf Groh
 * date:    17/02/2016
 * version: 1.0
 */
function Gehoelz(art, pflanzjahr, preis) {

	if (this.constructor === Gehoelz) {
        throw new Error("Fehler abstract class!");
      }

  /* Attribute der abstrakten Basisklasse */
  /* Attriubutes of the abstract base class */
  var art		= art;
  var pflanzjahr	= pflanzjahr;
  var preis		= preis;

  /* Get- und Set Methoden für alle Attribute der Klasse */
  /* Get and set methods for all attrubutes of the class */
  this.getArt = function() {
    return art;
  }

  this.setArt = function(art) {
    this.art = art;
  }

  this.getPflanzjahr = function() {
    return pflanzjahr;
  }

  this.setPflanzjahr = function(pflanzjahr) {
    this.pflanzjahr = pflanzjahr;
  }

  this.getPreis = function() {
    return preis;
  }

  this.setPreis = function(preis) {
    this.preis = preis;
  }

  /* Methode wird in allen Subklassen (ausser Liane) überschrieben */
  /* Method are overwritten in all (not in Liane) sub classes */
  this.getInfo = function() {
    return "Art: ".concat( art, " Pflanzjahr: ", pflanzjahr, " Preis: ", preis);
  }
}

File: LUA/Gehoelz.lua
--[[
 * autor:   Olaf Groh
 * date:    19/02/2016
 * version: 1.0
 ]]

Gehoelz = {}

  function Gehoelz.new(class, cart, cpflanzjahr, cpreis)


    if class==Gehoelz then
      error("Fehler abstract class!")
    end

    local self = {}

    --[[ Attribute der abstrakten Basisklasse ]]
    --[[ Attributes of the abstract base class ]]
    local art = cart
    local pflanzjahr = cpflanzjahr
    local preis = cpreis


  --[[ Get- und Set Methoden für alle Attribute der Klasse ]]
  --[[ Get and set methods for all attributes of the class ]]
  function self.getArt()
    return art
  end

  function self.setArt(sart)
    art = sart
  end

  function self.getPflanzjahr()
    return pflanzjahr
  end

  function self.setPflanzjahr(spflanzjahr)
    pflanzjahr = spflanzjahr
  end

  function self.getPreis()
    return preis
  end

  function self.setPreis(spreis)
    preis = spreis
  end

  --[[ Methode wird in allen Subklassen (ausser Liane) überschrieben ]]
  --[[ Method are overwritten in all (not in Liane) sub classes ]]
  function self.getInfo()
    return "Art: " .. art .. " Pflanzjahr: " .. pflanzjahr .. " Preis: " .. preis;
  end

  return self
end
